codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestJsonProtocolGenerator.java

```diff
@@ -174,14 +176,53 @@ private void writeShapeToJsonArray(
                 generateSimpleShapeToJsonValue(model, writer, memberShape, operand, (w, s) -> w.write("av.$L", s));
             }
         });
-	}
+    }
 
     @Override
     public void generateProtocolTests(GenerationContext context) {
+        Set<HttpProtocolUnitTestGenerator.ConfigValue> configValues = new TreeSet<>();
+        configValues.addAll(SetUtils.of(
+                HttpProtocolUnitTestGenerator.ConfigValue.builder()
+                        .name(AddAwsConfigFields.REGION_CONFIG_NAME)
+                        .value(writer -> {
+                            writer.writeInline("$S", "us-west-2");
+                        })
+                        .build(),
+                HttpProtocolUnitTestGenerator.ConfigValue.builder()
+                        .name(AddAwsConfigFields.ENDPOINT_RESOLVER_CONFIG_NAME)
+                        .value(writer -> {
+                            writer.addUseImports(AwsGoDependency.AWS_CORE);
+                            writer.openBlock("aws.EndpointResolverFunc("
+                                            + "func(service, region string) (e aws.Endpoint, err error) {",
+                                    "}),", () -> {
+                                        writer.write("e.URL = \"https://127.0.0.1\"");
+                                        writer.write("e.SigningRegion = \"us-west-2\"");
+                                        writer.write("return e, err");
+                                    });
+                        })
+                        .build(),
+                HttpProtocolUnitTestGenerator.ConfigValue.builder()
+                        .name("APIOptions")
+                        .value(writer -> {
+                            writer.addUseImports(SmithyGoDependency.SMITHY_MIDDLEWARE);
+                            writer.openBlock("[]APIOptionFunc{", "},", () -> {
```

Comment:
This most likely should append to the APIOptions field value instead of setting it.

Reply:
Ah is see, this is when the field member is set, not as a functional option.  NM this comment